### DISCUSSION TITLE
【V5】tickMethod 返回的 start 和 end 计算有误

### DISCUSSION THREAD

**Discussion by BinghuiXie** [antvis/G2#5329] [Q&A] [answered]
- **G2 Version**: 5.0.16
- **Platform**:
- **Mini Showcase(like screenshots)**:
- **CodePen Link**:

`tickMethod` 的 start 和 end 的计算有问题，在比例尺的类型是 `band` 或者 `point` 的时候，返回都是 `undefined`
![image](https://github.com/antvis/G2/assets/34343589/becc670c-f013-4d59-acf5-1963deab4837)

![image](https://github.com/antvis/G2/assets/34343589/0a3dfa96-8727-42ee-8d35-69675b3886a8)
而且这里能直接转数字吗？如果是字符串，得到的就是 NaN

**Comment by pearmini** [comment, marked as answer]
目前 G2 的约定是离散比例尺（ordinal 或者 point）没有或者不需要 tickMethod，因为它们的 tick 就是 domain。tickMethod 只适用于连续比例尺，如果对离散轴的 tick 有定制化需求，可以使用 labelFormatter 和 labelFilter 属性。

**Reply by BinghuiXie** [reply]
多请教一下，band、point 和 ordinal 有什么更形象的区别吗，看官网的描述感觉有一点抽象

**Reply by pearmini** [reply]
这三个定义域都是离散的，但是 band 和 point 的值域是连续的，主要用于映射位置通道；ordinal 的值域是离散的，常常用于映射颜色通道。point 是一种特殊的 band 比例尺，bandWidth 为 0。

可以看看这里的文档：https://github.com/antvis/scale/tree/master/docs/api#discrete
